src/test/java/org/littil/api/contact/api/ContactResourceTest.java

```diff
@@ -0,0 +1,131 @@
+import org.littil.api.auth.service.AuthenticationService;
+import org.littil.api.contact.service.Contact;
+import org.littil.api.user.service.User;
+import org.littil.api.user.service.UserService;
+import org.littil.mock.auth0.APIManagementMock;
+import org.littil.mock.coordinates.service.WireMockSearchService;
+
+import java.util.Optional;
+import java.util.UUID;
+
+import static io.restassured.RestAssured.given;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.doReturn;
+
+@QuarkusTest
+@TestHTTPEndpoint(ContactResource.class)
+@QuarkusTestResource(APIManagementMock.class)
+@QuarkusTestResource(WireMockSearchService.class)
+class ContactResourceTest {
+    @InjectSpy
+    UserService userService;
+    @InjectMock
+    TokenHelper tokenHelper;
+    @InjectMock
+    AuthenticationService authenticationService;
+
+    @Test
+    void givenFindAllUnauthorized_thenShouldReturnForbidden() {
+        given() //
+                .when() //
+                .get() //
+                .then() //
+                .statusCode(401);
+    }
+
+    @Test
+    @TestSecurity(user = "littil", roles = "viewer")
+    @OidcSecurity(claims = {
+            @Claim(key = "https://littil.org/littil_user_id", value = "0ea41f01-cead-4309-871c-c029c1fe19bf") })
+    void givenFindAll_thenShouldReturnOk() {
+        given() //
+                .when() //
+                .get() //
+                .then() //
+            .statusCode(200);
+    }
+
+    @Test
+    @TestSecurity(user = "littil", roles = "viewer")
+    @OidcSecurity(claims = {
+            @Claim(key = "https://littil.org/littil_user_id", value = "0ea41f01-cead-4309-871c-c029c1fe19bf") })
+    void givenCreateNewContact_thenShouldReturnCreatedSuccessfully() {
+        var contact = getContactPostResource(UUID.fromString("0ea41f01-cead-4309-871c-c029c1fe19bf"),"michael@littil.org","Berichtje");
+        var saved = sendAndSave(contact);
+
+        assertThat(saved.getStatusCode()).isEqualTo(200);
+    }
+
+    @Disabled("cant get this working (yet)")
+    @Test
+    @TestSecurity(user = "littil", roles = "viewer")
+    @OidcSecurity(claims = {
+            @Claim(key = "https://littil.org/littil_user_id", value = "0ea41f01-cead-4309-871c-c029c1fe19bf") })
+    void givenACreatedContactFindAll_thenShouldReturnAList() {
+        var contact = getContactPostResource(UUID.fromString("0ea41f01-cead-4309-871c-c029c1fe19bf"),"michael@littil.org","Berichtje");
+        var saved = sendAndSave(contact);
+        doReturn(Optional.of(UUID.fromString("0ea41f01-cead-4309-871c-c029c1fe19bf")))
+                .when(tokenHelper).currentUserId();
+        //FIXME: how to test? if we invoke findAllByCreatedByOrRecipientId directly in ContactResource with provided recipient id it does return values
```

Comment:
Dit komt waarschijnlijk omdat id met een random UUID wordt gevuld in de contactService